Diagnostic/Utils/misc_helpers.py

```diff
@@ -150,14 +150,12 @@ def log_and_report_failed_config_generation(self, ext_event_type, config_invalid
                     secrets in the protected settings. This is for logging to Geneva for diagnostic purposes.
         :return: None
         """
-        config_invalid_log = "Invalid config settings given: " + config_invalid_reason + \
-                             ". Can't proceed, although this install/enable operation is reported as successful so " \
-                             "the VM can complete successful startup."
-        self._logger_log(config_invalid_log)
-        self._status_reporter(ext_event_type, 'success', '0', config_invalid_log)
+        config_invalid_log = "Invalid config settings given: {0}".format(config_invalid_reason)
+        self._logger_error(config_invalid_log)
+        self._status_reporter(ext_event_type, 'error', '1', config_invalid_log)
```

Comment:
In the functions that originally returned errors we just returned '1' for all failure states, so I was worried about adding in more error codes in case waagent / portal only expect '0' or '1'. However, if you think adding a '2' for user-generated errors might be useful (to differentiate from system-generated errors), that could work?

Reply:
Updated user-generated errors to return '2'